Add update coordinator to Netgear LTE

tkdrob opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add update coordinator to Netgear LTE

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

tkdrob committed:
Add update coordinator to Netgear LTE

tkdrob committed:
Merge branch 'dev' of https://github.com/home-assistant/core into netgear_lte_coordinator

joostlek commented on homeassistant/components/netgear_lte/__init__.py:
```diff
@@ -212,33 +161,33 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     password = entry.data[CONF_PASSWORD]
 
-    if DOMAIN not in hass.data:
+    if not (data := hass.data.setdefault(DOMAIN, {})):
```
Why do we need this if statement?

tkdrob replied:
We only need one web session for the integration. Multiple entries can use the same one. Do you prefer it the old way?

joostlek commented on homeassistant/components/netgear_lte/__init__.py:
```diff
@@ -212,33 +161,33 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    hass.data[DOMAIN][entry.entry_id] = coordinator
 
-    await async_setup_services(hass)
+    await async_setup_services(hass, modem)
```
Can we type this?

joostlek commented on homeassistant/components/netgear_lte/entity.py:
```diff
@@ -1,54 +1,33 @@
         self.entity_description = description
-        self.modem_data = modem_data
-        self._attr_unique_id = f"{description.key}_{modem_data.data.serial_number}"
+        self._attr_unique_id = f"{description.key}_{coordinator.data.serial_number}"
```
You're using coordinator.data quite a lot, maybe you can use a local variable to avoid accessing the variable a lot

tkdrob replied:
Just for aesthetics? It doesn't save any lines by doing that.

joostlek replied:
It doesn't save lines but it looks better and is a bit more performant

tkdrob committed:
coordinator data tweak

tkdrob committed:
Merge branch 'dev' into netgear_lte_coordinator

frenck committed:
Bump version to 2024.4.0b0

TheJulianJES committed:
Bump zha-quirks to 0.0.113 (#114311)

edenhaus committed:
Remove checked in translations (#114336)

marcelveldt committed:
Fix Matter airconditioner discovery of climate platform (#114326)

* Discover Thermostat platform for Room Airconditioner device

* add test

* Adjust docstring

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

edenhaus committed:
Download translations only once in the build pipeline (#114335)

synesthesiam committed:
Add more Ollama models (#114339)

Add more models

cereal2nd committed:
Bump pyduotecno to 2024.3.2 (#114320)

emontnemery committed:
Don't access States.last_reported_ts before it's added (#114333)

bdraco committed:
Fix empty delays in script helper (#114346)

fixes
```
Logger: homeassistant.components.automation.kamermaster_knop_4_acties_licht
Bron: components/automation/__init__.py:726
integratie: Automatisering (documentatie, problemen)
Eerst voorgekomen: 22:17:29 (5 gebeurtenissen)
Laatst gelogd: 22:59:24

While executing automation automation.kamermaster_knop_4_acties_licht
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 726, in async_trigger
    return await self.action_script.async_run(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1645, in async_run
    return await asyncio.shield(create_eager_task(run.async_run()))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 454, in async_run
    await self._async_step(log_exceptions=False)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 506, in _async_step
    self._handle_exception(
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 536, in _handle_exception
    raise exception
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 504, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 626, in _async_delay_step
    if timeout_future.done():
       ^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'done'
```

elupus committed:
Bump fjaraskupan to 2.3.0 (#114344)

Update fjarakupen to 2.3.0

- Support delayed disconnection
- Speed up on/off transitions

bdraco committed:
Revert velocity change in powerview (#114337)

edenhaus committed:
Fix script for checking on existing translations (#114354)

emontnemery committed:
Improve utility meter restore state tests (#114356)

joostlek committed:
Fix streamlabswater feedback (#114371)

joostlek committed:
Fix Suez water feedback (#114372)

joostlek committed:
Fix Swiss public transport feedback (#114373)

joostlek committed:
Fix Lupusec feedback (#114374)

cdce8p committed:
Update pytile to 2023.12.0 (#114370)

jpbede committed:
Use `setup_test_component_platform` helper for sensor entity component tests instead of `hass.components` (#114316)

* Use `setup_test_component_platform` helper for sensor entity component tests instead of `hass.components`

* Missing file

* Fix import

* Remove invalid device class

emontnemery committed:
Improve device class of utility meter (#114368)

bieniu committed:
Adapt Tractive integration the latest API changes (#114380)

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

oyvindwe committed:
Avoid changing local time on Nobø Ecohub (#114332)

nobo_hub: Pass timezone to avoid changing local time on Nobø Ecohub in handshake

cdce8p committed:
Update romy to 0.0.9 (#114360)

MartinHjelmare committed:
Use fallback voice for selected language in cloud (#114246)

Co-authored-by: Erik Montnemery <erik@montnemery.com>

frenck committed:
Fix hassfest service icons check for custom integrations (#114389)

frenck committed:
Fix area search for entities of devices (#114394)

synesthesiam committed:
Filter preferred TTS format options if not supported (#114392)

Filter preferred format options if not supported

Commit:
Bump pyunifiprotect to 5.1.2 (#114348)

cottsay committed:
Bump aioraven to 0.5.3 (#114397)

piitaya committed:
Update frontend to 20240328.0 (#114396)

frenck committed:
Bump version to 2024.4.0b1

bdraco committed:
Cleanup some plex tasks that delayed startup (#114418)

sdb9696 committed:
Bump python-ring-doorbell to 0.8.8 (#114431)

* Bump ring_doorbell to 0.8.8

* Fix intercom history test for new library version

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

ALERTua committed:
Add overkiz bottom tank water temperature and core control water temperature for Atlantic Water Heater (#114186)

* Adds bottom tank water temperature and core conrol water temperature sensors for Atlantic water heater

* Update homeassistant/components/overkiz/sensor.py

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

* Update homeassistant/components/overkiz/sensor.py

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

---------

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

ALERTua committed:
Add overkiz heating status, absence mode, and boost mode binary sensors for Atlantic Water Heater (#114184)

* Adds heating status, absense mode, and boost mode binary sensors for Atlantic water heater

* Renamed absence mode and boost mode binary sensors

* Update homeassistant/components/overkiz/binary_sensor.py

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

* Update homeassistant/components/overkiz/binary_sensor.py

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

* Update homeassistant/components/overkiz/binary_sensor.py

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

---------

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

Tronix117 committed:
Follow real AtlanticPassAPCZoneControlZone physical mode on Overkiz (HEAT, COOL or HEAT_COOL) (#111830)

* Support HEAT_COOL when mode is Auto on overkiz AtlanticPassAPCZoneControlZone

* Refactor ZoneControlZone to simplify usic by only using a single hvac mode

* Fix linting issues

* Makes more sense to use halves there

* Fix PR feedback

bdraco committed:
Avoid concurrent radio operations with powerview hubs (#114399)

Co-authored-by: kingy444 <toddlesking4@hotmail.com>

epenet committed:
Log warnings in Renault initialisation (#114445)

iMicknl committed:
Bump pyoverkiz to 1.13.9 (#114442)

StevenLooman committed:
Bump async-upnp-client to 0.38.3 (#114447)

ALERTua committed:
Add overkiz water targets temperature numbers for Atlantic water heater (#114185)

* Adds water targets temperature numbers for Atlantic water heater

* Update homeassistant/components/overkiz/number.py

Co-authored-by: Mick Vleeshouwer <mick@imick.nl>

* Update homeassistant/components/overkiz/number.py

Co-authored-by: Mick Vleeshouwer <mick@imick.nl>

* ruff formatting reverted

* Update homeassistant/components/overkiz/number.py

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

* Update homeassistant/components/overkiz/number.py

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

* changed command hardcode to a constant

---------

Co-authored-by: Mick Vleeshouwer <mick@imick.nl>
Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

piitaya committed:
Update frontend to 20240329.0 (#114452)

frenck committed:
Bump version to 2024.4.0b2

piitaya committed:
Update frontend to 20240329.1 (#114459)

bdraco committed:
Avoid tracking import executor jobs (#114453)

frenck committed:
Bump version to 2024.4.0b3

bdraco committed:
Mark executor jobs as background unless created from a tracked task (#114450)

* Mark executor jobs as background unless created from a tracked task

If the current task is not tracked the executor job should not
be a background task to avoid delaying startup and shutdown.

Currently any executor job created in a untracked task or
background task would end up being tracked and delaying
startup/shutdown

* import exec has the same issue

* Avoid tracking import executor jobs

There is no reason to track these jobs as they are always awaited
and we do not want to support fire and forget import executor jobs

* fix xiaomi_miio

* lots of fire time changed without background await

* revert changes moved to other PR

* more

* more

* more

* m

* m

* p

* fix fire and forget tests

* scrape

* sonos

* system

* more

* capture callback before block

* coverage

* more

* more races

* more races

* more

* missed some

* more fixes

* missed some more

* fix

* remove unneeded

* one more race

* two

thecode committed:
Cleanup Shelly RGBW light entities (#114410)

bdraco committed:
Improve sonos test synchronization (#114468)

bdraco committed:
Fix workday doing blocking I/O in the event loop (#114492)

bdraco committed:
Fix late load of anyio doing blocking I/O in the event loop (#114491)

* Fix late load of anyio doing blocking I/O in the event loop

httpx loads anyio which loads the asyncio backend in the event loop
as soon as httpx makes the first request

* tweak

synesthesiam committed:
Add initial support for floors to intents (#114456)

* Add initial support for floors to intents

* Fix climate intent

* More tests

* No return value

* Add requested changes

* Reuse event handler

dotvav committed:
Fix Overkiz Hitachi OVP air-to-air heat pump (#114487)

Unpack command parameters instead of passing a list

gjohansson-ST committed:
Fix server update from breaking setup in Speedtest.NET (#114524)

Kane610 committed:
Bump axis to v60 (#114544)

* Improve Axis MQTT support

* Bump axis to v60

cereal2nd committed:
Bump velbusaio to 2024.4.0 (#114569)

Bump valbusaio to 2024.4.0

joostlek committed:
Fix wrong icons (#114567)

* Fix wrong icons

* Fix wrong icons

jjlawren committed:
Bump plexapi to 4.15.11 (#114581)

mib1185 committed:
Reduce usage of executer threads in AVM Fritz!Tools (#114570)

* call entity state update calls in one executer task

* remove not needed wrapping

* mark as "non-public" method

* add guard against changes on _entity_update_functions

bdraco committed:
Bump zeroconf to 0.132.0 (#114596)

changelog: https://github.com/python-zeroconf/python-zeroconf/compare/0.131.0...0.132.0

IngoK1 committed:
Fix for Sonos URL encoding problem #102557 (#109518)

* Fix for URL encoding problem #102557

Fixes the problem "Cannot play media with spaces in folder names to Sonos #102557" removing the encoding of the strings in the music library.

* Fix type casting problem

* Update media_browser.py to fix pr check findings

Added required casting for all unquote statements to avoid further casting findings in the pr checks

* Update media_browser.py

Checked on linting, lets give it another try

* Update media_browser.py

Updated ruff run

* Update media_browser.py - added version run through ruff

* Update media_browser.py - ruff changes

* Apply ruff formatting

* Update homeassistant/components/sonos/media_browser.py

Co-authored-by: jjlawren <jjlawren@users.noreply.github.com>

* Update homeassistant/components/sonos/media_browser.py

Co-authored-by: jjlawren <jjlawren@users.noreply.github.com>

* Update homeassistant/components/sonos/media_browser.py

Co-authored-by: jjlawren <jjlawren@users.noreply.github.com>

* Update homeassistant/components/sonos/media_browser.py

Co-authored-by: jjlawren <jjlawren@users.noreply.github.com>

---------

Co-authored-by: computeq-admin <51021172+computeq-admin@users.noreply.github.com>
Co-authored-by: Jason Lawrence <jjlawren@users.noreply.github.com>

MartinHjelmare committed:
Fix reolink media source data access (#114593)

* Add test

* Fix reolink media source data access

bdraco committed:
Avoid storing raw extracted traceback in system_log (#114603)

This is never actually used and takes up quite a bit of ram

bdraco committed:
Fix memory leak when importing a platform fails (#114602)

* Fix memory leak when importing a platform fails

re-raising ImportError would trigger a memory leak

* fixes, coverage

* Apply suggestions from code review

bdraco committed:
Add missing platforms_exist guard to check_config (#114600)

* Add missing platforms_exist guard to check_config

related issue #112811

When the exception hits, the config will end up being saved in the traceback
so the memory is never released.

This matches the check_config code to homeassistant.config to avoid having
the exception thrown.

* patch

* merge branch

mkmer committed:
Bump whirlpool-sixth-sense to 0.18.7 (#114606)

Bump sixth-sense to 0.18.7

mib1185 committed:
Filter out ignored entries in ssdp step of AVM Fritz!SmartHome (#114574)

filter out ignored entries in ssdp step

balloob committed:
Bump version to 2024.4.0b4

max2697 committed:
Bump opower to 0.4.2 (#114608)

mkmer committed:
Add diagnostic platform to Whirlpool (#114578)

* Add diagnostic platform and tests

* lowercase variable

* Correc doc string

PeteRager committed:
Display sonos album title with URL encoding (#113693)

* unescape the title

When extracting the title from the item_id, it needs to be unescaped.

* sort imports

bieniu committed:
Improve Shelly RPC device update progress (#114566)

Co-authored-by: Shay Levy <levyshay1@gmail.com>
Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

frenck committed:
Add floor selector (#114614)

boswelja committed:
Catch potential ValueError when getting or setting Starlink sleep values (#114607)

joostlek committed:
Bump roombapy to 1.8.1 (#114478)

* Bump roombapy to 1.7.0

* Bump

* Bump

* Fix

Fexiven committed:
Fix Starlink integration startup issue (#114615)

bramkragten committed:
Update frontend to 20240402.0 (#114627)

Bre77 committed:
Fix battery heater in Tessie (#114568)

frenck committed:
Bump version to 2024.4.0b5

dotvav committed:
Fix Overkiz Hitachi OVP air-to-air heat pump (#114611)

sdb9696 committed:
Bump ring_doorbell integration to 0.8.9 (#114631)

bramkragten committed:
Update frontend to 20240402.1 (#114646)

puddly committed:
Reduce ZHA OTA logbook entries and extraneous updates (#114591)

frenck committed:
Bump version to 2024.4.0b6

bieniu committed:
Add missing state to the Tractive tracker state sensor (#114654)

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

Kane610 committed:
Bump aiounifi to v74 (#114649)

gjohansson-ST committed:
Bump holidays to 0.46 (#114657)

bdraco committed:
Avoid trying to load platform that are known to not exist in async_prepare_setup_platform (#114659)

PeteRager committed:
Fix Sonos play imported playlists (#113934)

bramkragten committed:
Update frontend to 20240402.2 (#114683)

frenck committed:
Bump version to 2024.4.0b7

dmulcahey committed:
Import zha quirks in the executor (#114685)

lellky committed:
Bump flexit_bacnet to 2.2.1 (#114641)

joostlek committed:
Fix Downloader config flow (#114718)

LennP committed:
Rename Motionblinds BLE integration to Motionblinds Bluetooth (#114584)

joostlek committed:
Avoid blocking IO in downloader config flow (#114741)

bramkragten committed:
Update frontend to 20240403.0 (#114747)

frenck committed:
Bump version to 2024.4.0b8

edenhaus committed:
Allow passing area/device/entity IDs to floor_id and floor_name (#114748)

synesthesiam committed:
Bump intents (#114755)

bramkragten committed:
Update frontend to 20240403.1 (#114756)

mib1185 committed:
Fix Synology DSM setup in case no Surveillance Station permission (#114757)

frenck committed:
Bump version to 2024.4.0b9

frenck committed:
Bump version to 2024.4.0

frenck committed:
2024.4.0 (#114764)

xeniter committed:
Update romy to 0.0.10 (#114785)

bachya committed:
Fix unhandled `KeyError` during Notion setup (#114787)

Noltari committed:
Update aioairzone-cloud to v0.4.7 (#114761)

cdheiser committed:
Fix Lutron light brightness values (#114794)

Fix brightness values in light.py

Bugfix to set the brightness to 0-100 which is what Lutron expects.

bbr111 committed:
Fix missing if statement in homematic (#114832)

* homematic fix issue #114807

Update climate.py

* Update homeassistant/components/homematic/climate.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

joostlek committed:
Avoid blocking IO in downloader initialization (#114841)

* Avoid blocking IO in downloader initialization

* Avoid blocking IO in downloader initialization

timmo001 committed:
Pin systembridgemodels to 4.0.4 (#114842)

bramkragten committed:
Update frontend to 20240404.0 (#114859)

bdraco committed:
Refactor ConfigStore to avoid needing to pass config_dir (#114827)

Co-authored-by: Erik <erik@montnemery.com>

tronikos committed:
Bump opower to 0.4.3 (#114826)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

bramkragten committed:
Update frontend to 20240404.1 (#114890)

astrandb committed:
Bump myuplink dependency to 0.6.0 (#114767)

bdraco committed:
Reduce august polling frequency (#114904)

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

jeeftor committed:
Bump weatherflow4py to 0.2.20 (#114888)

bdraco committed:
Ensure all tables have the default table args in the db_schema (#114895)

Kane610 committed:
Fix Axis camera platform support HTTPS (#114886)

lextm committed:
Fix type cast in snmp (#114795)

bdraco committed:
Handle ambiguous script actions by using action map order (#114825)

Kane610 committed:
Fix Axis reconfigure step not providing protocols as alternatives but as string (#114889)

astrandb committed:
Show correct model string in myuplink (#114921)

joostlek committed:
Fix Downloader YAML import (#114844)

joostlek committed:
Fix cast dashboard in media browser (#114924)

joostlek committed:
Create right import issues in Downloader (#114922)

* Create right import issues in Downloader

* Create right import issues in Downloader

* Create right import issues in Downloader

* Create right import issues in Downloader

* Fix

* Fix

* Fix

* Fix

* Apply suggestions from code review

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Fix

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

frenck committed:
Bump version to 2024.4.1

joostlek committed:
Fix ROVA validation (#114938)

* Fix ROVA validation

* Fix ROVA validation

frenck committed:
2024.4.1 (#114934)

joostlek committed:
Fix Snapcast Config flow (#114952)

Kane610 committed:
Bump axis to v61 (#114964)

oyvindwe committed:
Upgrade to pynobo 1.8.1 (#114982)

pynobo 1.8.1

bdraco committed:
Fix dictionary changed size during iteration in prometheus (#115005)

Fixes #104803

natekspencer committed:
Bump pylitterbot to 2023.4.11 (#114918)

bieniu committed:
Bump `brother` to version 4.1.0 (#115021)

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

farmio committed:
Update xknxproject to 3.7.1 (#115053)

bdraco committed:
Fix synology_dsm availablity (#115073)

* Remove reload on update failure from synology_dsm

fixes #115062

The coordinator will retry on its own later, there is no
reason to reload here.  This was added in #42697

* fix available checks

ludeeus committed:
Downgrade hass-nabucasa from 0.80.0 to 0.78.0 (#115078)

cereal2nd committed:
Bump velbus-aio to 2024.4.1 (#115109)

bump velbusaio to 2024.4.1

dontinelli committed:
improve handling of incorrect values in fyta integration (#115134)

* improve handling of incorrect values

* Changes based on review comment

* Apply suggestions from code review

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* update value_fn

* ruff

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

dontinelli committed:
Bump fyta_cli to 0.3.5 (#115143)

bump fyta_cli to 0.3.5

rappenze committed:
Fix fibaro sensor additional sensor lookup (#115148)

bdraco committed:
Terminate scripts with until and while conditions that execute more than 10000 times (#115110)

matrixd2 committed:
Bump yolink-api to 0.4.2 (#115026)

bdraco committed:
Write timer entity state before firing events (#115151)

bdraco committed:
Avoid checking for polling if an entity fails to add (#115159)

* Avoid checking for polling if an entity fails to add

* no need to do protected access

* no need to do protected access

* no need to do protected access

* no need to do protected access

* coverage

* fix test

* fix

* broken one must be first

joostlek committed:
Filter out fuzzy translations from Lokalise (#114968)

nmaggioni committed:
Fix hang in SNMP device_tracker implementation (#112815)

Co-authored-by: J. Nick Koston <nick@koston.org>

gibwar committed:
Only reset requested utility meter with no tariff (#115170)

frenck committed:
Bump version to 2024.4.2

frenck committed:
2024.4.2 (#115186)

bdraco committed:
Fix misssing timeout in caldav (#115247)

allenporter committed:
Fix Google Tasks parsing of remove responses (#115258)

OnFreund committed:
Configurable maximum concurrency in Risco local (#115226)

* Configurable maximum concurrency in Risco local

* Show advanced Risco options in advanced mode

janiversen committed:
Bump pymodbus v3.6.7 (#115279)

Bump pymodbus v3.6.7.

klaasnicolaas committed:
Bump forecast-solar lib to v3.1.0 (#115272)

joostlek committed:
Bump yt-dlp to 2024.04.09 (#115295)

TheJulianJES committed:
Bump zha-quirks to 0.0.114 (#115299)

bdraco committed:
Bump bleak-retry-connector 3.5.0 (#115328)

thecode committed:
Fix Aranet failure when the Bluetooth proxy is not providing a device name (#115298)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Ensure automations do not execute from a trigger if they are disabled (#115305)

* Ensure automations are stopped as soon as the stop future is set

* revert script changes and move them to #115325

janiversen committed:
Solve modbus test problem (#115376)

Fix test.

janiversen committed:
Secure against resetting a non active modbus (#115364)

OnFreund committed:
Improve Risco exception logging (#115232)

bdraco committed:
Fix deadlock in holidays dynamic loading (#115385)

mdegat01 committed:
Add support for adopt data disk repair (#114891)

NodeJSmith committed:
Bump whirlpool-sixth-sense to 0.18.8 (#115393)

bump whirlpool to 0.18.8

Santobert committed:
Bump pybotvac to 0.0.25 (#115435)

Bump pybotvac

bdraco committed:
Bump aiohttp to 3.9.4 (#110730)

* Bump aiohttp to 3.9.4

This is rc0 for now but will be updated when the full release it out

* cleanup cruft

* regen

* fix tests (these changes are fine)

* chunk size is too small to read since boundry is now enforced

* chunk size is too small to read since boundry is now enforced

allenporter committed:
Fix bug in rainbird switch when turning off a switch that is already off (#115421)

Fix big in rainbird switch when turning off a switch that is already off

Co-authored-by: J. Nick Koston <nick@koston.org>

agners committed:
Support backup of add-ons with hyphens (#115274)

Co-authored-by: J. Nick Koston <nick@koston.org>

bramkragten committed:
Update frontend to 20240404.2 (#115460)

frenck committed:
Bump version to 2024.4.3

frenck committed:
2024.4.3 (#115463)

tkdrob committed:
Merge branch 'master' of https://github.com/home-assistant/core into netgear_lte_coordinator

tkdrob committed:
finish merge conflict correction
